tribe-ext-relabeler/src/Tribe/Settings_Helper.php

```diff
@@ -0,0 +1,171 @@
+		if ( ! is_string( $neighboring_field ) ) {
+			// If neighbor is not specified, add this to misc section.
+			$this->insert_fields_misc = array_replace_recursive(
+				$this->insert_fields_misc,
+				array( $setting_tab => $fields )
+			);
+		} elseif ( true === $above ) {
+			// Add to above fields list with neighbor specified.
+			$this->insert_fields_above = array_replace_recursive(
+				$this->insert_fields_above,
+				array( $setting_tab => array( $neighboring_field => $fields ) )
+			);
+		} else {
+			// Add to below fields list with neighbor specified.
+			$this->insert_fields_below = array_replace_recursive(
+				$this->insert_fields_below,
+				array( $setting_tab => array( $neighboring_field => $fields ) )
+			);
+		}
+
+	}
+
+
+	/**
+	 * Remove a field from one of the tabs in WP Admin > Events > Settings
+	 *
+	 * @param string $field_key   Option key for your setting. Example: 'fancyOptionName'.
+	 * @param string $setting_tab Settings tab where this will be added. Example: 'display'.
+	 */
+	public function remove_field( $field_key, $setting_tab ) {
+		$this->remove_fields[ $setting_tab ][] = $field_key;
+	}
+
+
+	/**
+	 * Attached to 'tribe_settings_tab_fields' to add/remove this class' fields on Tribe Settings pages.
+	 *
+	 * @param array $fields The fields within tribe settings page.
+	 * @param string $tab   The settings tab key.
+	 *
+	 * @return array $fields The fields within tribe settings page
+	 */
+	public function filter_options( $fields, $tab ) {
+
+		// Fields appended to misc section.
+		if ( array_key_exists( $tab , $this->insert_fields_misc ) ) {
+
+			// Add a misc heading if none exists.
+			if ( ! array_key_exists( 'tribeMiscSettings', $fields ) ) {
+				$misc_heading = array(
+					'tribeMiscSettings' => array(
+						'type' => 'html',
+						'html' => '<h3>' . esc_html__( 'Miscellaneous Settings', 'tribe-common' ) . '</h3>',
+					),
+				);
+				$fields = Tribe__Main::array_insert_before_key( 'tribe-form-content-end', $fields, $misc_heading );
+			}
+
+			// Insert these settings under misc heading.
+			$fields = Tribe__Main::array_insert_after_key( 'tribeMiscSettings', $fields, $this->insert_fields_misc[ $tab ] );
+		}
+
+		// Fields inserted above a neighboring field.
+		if ( array_key_exists( $tab , $this->insert_fields_above ) ) {
+
+			foreach ( $this->insert_fields_above[ $tab ] as $insert_after => $new_field ) {
+				$fields = Tribe__Main::array_insert_before_key( $insert_after, $fields, $new_field );
+			}
+		}
+
+		// Fields inserted below a neighboring field.
+		if ( array_key_exists( $tab , $this->insert_fields_below ) ) {
+
+			foreach ( $this->insert_fields_below[ $tab ] as $insert_after => $new_field ) {
+				$fields = Tribe__Main::array_insert_after_key( $insert_after, $fields, $new_field );
+			}
+		}
+
+		// Fields that will be removed.
+		if ( array_key_exists( $tab , $this->remove_fields ) ) {
```

Comment:
Nitpick: here and _also_ on lines 124, 142, 150, why the extra space around comma?

Instead of this, I mean:
`if ( array_key_exists( $tab , $this->remove_fields ) )`

I would recommend this, if there's not an explicit reason for the above:
`if ( array_key_exists( $tab, $this->remove_fields ) )`